packages/server-common/src/handlers/user/invite.handler.ts

```diff
@@ -15,73 +11,69 @@ import {
 } from '../../utils';
 import { UserPayload } from '../event-bus';
 
-const uuidMatch =
-  /^([\d\w]{8})-?([\d\w]{4})-?([\d\w]{4})-?([\d\w]{4})-?([\d\w]{12})|[{0x]*([\d\w]{8})[0x, ]{4}([\d\w]{4})[0x, ]{4}([\d\w]{4})[0x, {]{5}([\d\w]{2})[0x, ]{4}([\d\w]{2})[0x, ]{4}([\d\w]{2})[0x, ]{4}([\d\w]{2})[0x, ]{4}([\d\w]{2})[0x, ]{4}([\d\w]{2})[0x, ]{4}([\d\w]{2})[0x, ]{4}([\d\w]{2})$/;
+interface DataProvider {
+  fetchById(id: string): Promise<gp2.UserDataObject | UserDataObject | null>;
+  update(
+    id: string,
+    user: gp2.UserUpdateDataObject | UserUpdateDataObject,
+  ): Promise<void>;
+}
 
 export const inviteHandlerFactory =
-  (
+  <Provider extends DataProvider>(
     sendEmail: SendEmail,
-    userClient: SquidexRestClient<RestUser | gp2Squidex.RestUser>,
+    dataProvider: Provider,
     origin: string,
     logger: Logger,
     template: SendEmailTemplate = 'Crn-Welcome',
   ): EventBridgeHandler<'UsersPublished', UserPayload> =>
   async (event) => {
-    let user: RestUser | gp2Squidex.RestUser;
-
-    try {
-      user = await userClient.fetchById(event.detail.payload.id);
-    } catch (error) {
-      logger.error(error, 'Error while fetching user');
+    const user = await dataProvider.fetchById(event.detail.resourceId);
+    if (!user) {
       throw new Error(
-        `Unable to find a user with ID ${event.detail.payload.id}`,
+        `Unable to find a user with ID ${event.detail.resourceId}`,
       );
     }
 
-    logger.debug(
-      `Attempting to invite user with ID ${event.detail.payload.id}, e-mail address ${user.data.email}`,
-    );
-
-    const previousCode = user.data.connections.iv
-      ?.map((c) => c.code)
-      .find((c) => c.match(uuidMatch));
-
-    if (previousCode) {
+    if (user.connections?.length) {
       logger.info(
         `Found a previous invitation code for user ${user.id}, exiting...`,
       );
       return;
     }
-    const newCode = uuidV4();
+
+    logger.debug(
+      `Attempting to invite user with ID ${event.detail.resourceId}, e-mail address ${user.email}`,
+    );
+
+    const code = uuidV4();
 
     try {
-      await userClient.patch(user.id, {
-        connections: {
-          iv: [...(user.data.connections.iv || []), { code: newCode }],
-        },
+      await dataProvider.update(user.id, {
```

Comment:
alright this one was actually here for a reason. See - we had this issue whereby sometimes squidex would send out the 'user-create' event for all dev users. Although the actually root cause of it was the order in which the data was impoted into the PR environments we made sure that the previous connections would not be removed when such invite handler was fired. 

I understand we want to skip creating connections for users that already have ANY connection code (unlike previously when we would only skip if they had an invitation code) but this line here - i don't think it needs to be modified? In which case you should make sure that we get the current user connection codes and add the new one

Reply:
By this point we have already checked that there are no existing user connection codes and exited early if any exist. 

`...(user.data.connections.iv || [])` is equivalent to `...[]` here, and so was removed.